woocommerce/woocommerce-gutenberg-products-block
# Remove state from useCheckoutAddress

Some changes to `useCheckoutAddress` to remove its state and complex billing address handling.

- Removes state from `useCheckoutAddress`. Handling for the shipping as billing field is moved to `CustomerDataContext`
- `CustomerDataContext` converted to TypeScipt.
- Fixes initial value of `shippingAsBilling` by waiting for `useCustomerData` to be Initialized. Before it was comparing two empty addresses, so the checkbox would never be left unchecked.

Fixes #5803

### Testing

How to test the changes in this Pull Request:

1. In a new window logged out, add something to cart and go to checkout
2. Enter a shipping address
3. Uncheck "use shipping as billing"
4. Enter a different billing address
5. Place order
6. Add another item to cart and checkout again
7. "use shipping as billing" should be unchecked and you should see the different shipping and billing addresses
8. Check "use shipping as billing"
9. Uncheck "use shipping as billing"
10. Billing should be the same as shipping (previous values removed)

After testing, check the orders in WordPress admin and confirm the correct addresses were persisted.

### Changelog

> Toggling the "use shipping as billing" checkbox will now copy values from shipping rather than preserve a past entered billing address.
